include/aos/common/error.hpp

```diff
@@ -72,6 +72,38 @@ class Error : public EnumStringer<ErrorType> {
     {
     }
 
+    /**
+     * Constructs error instance from another error.
+     */
+    Error(const Error& err, const char* fileName = nullptr, int lineNumber = 0)
+        : EnumStringer(err.GetValue())
+        , mErrno(err.mErrno)
+        , mFileName(fileName)
+        , mLineNumber(lineNumber)
+    {
+        if (!mFileName) {
+            mFileName = err.mFileName;
+            mLineNumber = err.mLineNumber;
+        }
+    }
+
+    /**
+     * Assigns error from another error.
+     *
+     * @param err error to copy from.
+     */
+    Error& operator=(const Error& err)
+    {
```

Comment:
If we have a derived class, then in the copy constructor as well as in the copy assignment operator, we need to invoke the copy assignment operator for the base class. In this case, it would result in the subobject of the base class has an incorrect value.
```
EnumStringer::operator=(err.GetValue())
```

Reply:
You are right. Fixed.